standard_sequence/experiment_components.py

```diff
@@ -0,0 +1,816 @@
+        devices.tweezer_aom_analog.ramp(
+            t, duration=dur, initial=self.tw_power, final=final_power, samplerate=1e5
+        )
+        self.tw_power = final_power
+        return t + dur
+
+    def sine_mod_power(self, t, dur, amp, freq):
+        devices.tweezer_aom_analog.sine(
+            t,
+            duration=dur,
+            amplitude=amp,
+            angfreq=2 * np.pi * freq,
+            phase=0,
+            dc_offset=self.tw_power,
+            samplerate=1e5,
+        )
+
+
+class Microwave:
+    def __init__(self, t):
+        CONST_SPECTRUM_UWAVE_CABLE_ATTEN = (
+            4.4  # cable attenutation, dB  meausred at 300 MHz
+        )
+        self.mw_detuning = shot_globals.mw_detuning  # tune the microwave detuning here
+        self.uwave_dds_switch_on = True
+        self.uwave_absorp_switch_on = False
+        self.spectrum_uwave_power = -1  # dBm power set at the input of dds switch, this power is set to below the amplifier damage threshold
+        devices.uwave_dds_switch.go_high(
+            t
+        )  # dds_switch always on, can be off if need further higher distinguishing ratio
+        devices.uwave_absorp_switch.go_low(
+            t
+        )  # absorp switch only on when sending pulse
+        devices.spectrum_uwave.set_mode(
+            replay_mode=b"sequence",
+            channels=[
+                {
+                    "name": "microwaves",
+                    "power": self.spectrum_uwave_power
+                    + CONST_SPECTRUM_UWAVE_CABLE_ATTEN,
+                    "port": 0,
+                    "is_amplified": False,
+                    "amplifier": None,
+                    "calibration_power": 12,
+                    "power_mode": "constant_total",
+                    "max_pulses": 1,
+                },
+                {
+                    "name": "mmwaves",
+                    "power": -11,
+                    "port": 1,
+                    "is_amplified": False,
+                    "amplifier": None,
+                    "calibration_power": 12,
+                    "power_mode": "constant_total",
+                    "max_pulses": 1,
+                },
+            ],
+            clock_freq=625,
+            use_ext_clock=True,
+            ext_clock_freq=10,
+        )  # spectrum setup for microwaves & mmwaves, 1st channel for Hyperfine splitting of ground state, 2nd channel for mmwaves on Rydberg levels
+
+    def do_pulse(self, t, dur):
+        """do microwave pulse"""
+        CONST_SPECTRUM_CARD_OFFSET = (
+            52.8e-6  # the delay time between spectrum card output and trigger
+        )
+
+        t += CONST_SPECTRUM_CARD_OFFSET
+        devices.uwave_absorp_switch.go_high(t)
+        self.uwave_absorp_switch_on = True
+        devices.spectrum_uwave.single_freq(
+            t - CONST_SPECTRUM_CARD_OFFSET,
+            duration=dur,
+            freq=spec_freq_calib(self.mw_detuning),
+            amplitude=0.99,

```

Comment:
brief comment about why 0.99 is preferred instead of "1"

Reply:
Added